iotfunctions/db.py

```diff
@@ -429,7 +442,47 @@ def drop_table(self,table_name,schema=None):
             msg = 'Dropped table name %s' %table.name
             self.session.commit()
         logger.debug(msg)
-               
+        
+    def get_entity_type(self,name):
+        '''
+        Get an EntityType instance by name. Name may be the logical name shown in the UI or the table name.'
+        
+        '''
+        metadata = None
+        try:
+            metadata = self.entity_type_metadata[name]
+        except KeyError:
+            for m in list(self.entity_type_metadata.values()):
+                if m['metricTableName'] == name:
+                    metadata = m
+                    break
+            msg = 'No entity called % in the cached metadata.' %name
+            raise ValueError(msg)
+            
+        print (metadata)
+        raise
+                
+        timestamp = metadata['metricTimestampColumn']
+        schema = metadata['schemaName']
+        dim_table = metadata['dimensionTableName']
+        try:
+            entity_type_id = metadata['entityTypeId']   
+        except KeyError:
+            entity_type_id = None   

```

Comment:
entity_type_id = metadata.get('entityTypeId', None)

Reply:
Thanks. That is cleaner.